tests/integration/test_author_api.py

```diff
@@ -0,0 +1,128 @@
+    The functionality is provided by invenio-records-rest, which is not the
+    subject of these tests.
+    """
+    with app.test_client() as client:
+        assert client.get(
+            '/api/authors/983220').status_code == 200
+
+        request = json.loads(
+            client.get('/api/authors/983220').data)
+        print(request)
+        assert request['id'] == 983220
+
+
+def test_author_citations_endpoint(app):
+    """This test checks for the total count of objects and the availability
+    of required keys and their type.
+    """
+    with app.test_client() as client:
+        assert client.get(
+            '/api/authors/983220/citations/').status_code == 200
+
+        request = json.loads(
+            client.get('/api/authors/983220/citations/').data)
+
+        assert len(request) == 197
+        assert isinstance(request[0]['date'], six.string_types)
+        assert isinstance(request[0]['self_citation'], bool)
+        assert isinstance(request[0]['citee']['record']['$ref'],
+                          six.string_types)
+        assert isinstance(request[0]['citer']['record']['$ref'],
+                          six.string_types)
+        assert isinstance(request[0]['published_paper'], bool)
+
+
+def test_author_coauthors_endpoint(app):
+    """This test checks for the total count of objects and the availability
+    of required keys and their type.
+    """
+    with app.test_client() as client:
+        assert client.get(
+            '/api/authors/983220/coauthors/').status_code == 200
+
+        request = json.loads(
+            client.get('/api/authors/983220/coauthors/').data)
+
+        assert len(request) == 100
+        assert isinstance(request[0]['count'], int)
+        assert isinstance(request[0]['record']['$ref'], six.string_types)
+        assert isinstance(request[0]['id'], int)
+        assert isinstance(request[0]['full_name'], six.string_types)
+
+
+def test_author_publications_endpoint(app):
+    """This test checks for the total count of objects and the availability
+    of required keys and their type.
+    """
+    with app.test_client() as client:
+        assert client.get(
+            '/api/authors/983220/publications/').status_code == 200
+
+        request = json.loads(
+            client.get('/api/authors/983220/publications/').data)
+
+        assert len(request) == 30
+        assert isinstance(request[0]['title'], six.string_types)
+        assert isinstance(request[0]['collaborations'][0], six.string_types)
+        assert isinstance(request[0]['journal']['record']['$ref'],
+                          six.string_types)
+        assert isinstance(request[0]['record']['$ref'], six.string_types)
+        assert isinstance(request[0]['citations'], int)
+        assert isinstance(request[0]['date'], six.string_types)
+        assert isinstance(request[0]['type'], six.string_types)
+        assert isinstance(request[0]['id'], int)
```

Comment:
Idea: use JSON Schema to specify the format and the types of the response, and assert that the response respects the schema (this is actually how most people use JSON Schema).